pkg/drivers/kic/kic.go

```diff
@@ -0,0 +1,280 @@
+	}
+	if o == "restarting" {
+		return state.Starting, nil
+	}
+	if o == "dead" {
+		return state.Error, nil
+	}
+	return state.None, fmt.Errorf("unknown state")
+
+}
+
+// Kill stops a host forcefully, including any containers that we are managing.
+func (d *Driver) Kill() error {
+	cmd := exec.Command(d.NodeConfig.OCIBinary, "kill", d.MachineName)
+	if err := cmd.Run(); err != nil {
+		return errors.Wrapf(err, "killing kic node %s", d.MachineName)
+	}
+	return nil
+}
+
+// Remove will delete the Kic Node Container
+func (d *Driver) Remove() error {
+	if _, err := d.nodeID(d.MachineName); err != nil {
+		return errors.Wrapf(err, "not found node %s", d.MachineName)
+	}
+	cmd := exec.Command(d.NodeConfig.OCIBinary, "rm", "-f", "-v", d.MachineName)
+	if err := cmd.Run(); err != nil {
+		return errors.Wrapf(err, "error removing node %s", d.MachineName)
+	}
+	return nil
+}
+
+// Restart a host
+func (d *Driver) Restart() error {
+	s, err := d.GetState()
+	if err != nil {
+		return errors.Wrap(err, "get kic state")
+	}
+	switch s {
+	case state.Paused:
+		return d.Unpause()
+	case state.Stopped:
+		return d.Start()
+	case state.Running, state.Error:
+		if err = d.Stop(); err != nil {
+			return fmt.Errorf("restarting a kic stop phase %v", err)
+		}
+		if err = d.Start(); err != nil {
+			return fmt.Errorf("restarting a kic start phase %v", err)
+		}
+		return nil
+	}
+
+	return fmt.Errorf("restarted not implemented for kic state %s yet", s)
+}
+
+// Unpause a kic container
+func (d *Driver) Unpause() error {
+	cmd := exec.Command(d.NodeConfig.OCIBinary, "pause", d.MachineName)
+	if err := cmd.Run(); err != nil {
+		return errors.Wrapf(err, "unpausing %s", d.MachineName)
+	}
+	return nil
+}
+
+// Start a _stopped_ kic container
+// not meant to be used for Create().
+func (d *Driver) Start() error {
+	s, err := d.GetState()
+	if err != nil {
+		return errors.Wrap(err, "get kic state")
+	}
+	if s == state.Stopped {
+		cmd := exec.Command(d.NodeConfig.OCIBinary, "start", d.MachineName)
+		if err := cmd.Run(); err != nil {
+			return errors.Wrapf(err, "starting a stopped kic node %s", d.MachineName)
+		}
+		return nil
+	}
+	return fmt.Errorf("cant start a not-stopped (%s) kic node", s)
+}
+
+// Stop a host gracefully, including any containers that we are managing.
+func (d *Driver) Stop() error {
+	cmd := exec.Command(d.NodeConfig.OCIBinary, "stop", d.MachineName)
+	if err := cmd.Run(); err != nil {
+		return errors.Wrapf(err, "stopping %s", d.MachineName)
+	}
+	return nil
+}
+
+// RunSSHCommandFromDriver implements direct ssh control to the driver
+func (d *Driver) RunSSHCommandFromDriver() error {
+	return fmt.Errorf("driver does not support RunSSHCommandFromDriver commands")
+}
+
+// looks up for a container node by name, will return error if not found.
+func (d *Driver) nodeID(nameOrID string) (string, error) {
+	cmd := exec.Command(d.NodeConfig.OCIBinary, "inspect", "-f", "{{.Id}}", nameOrID)
+	id, err := cmd.CombinedOutput()
+	if err != nil {
```

Comment:
Do we need this error check if we return err later?

Reply:
good call !